src/time/src/mcp_server_time/server.py

```diff
@@ -0,0 +1,145 @@
+from datetime import datetime
+import json
+from typing import Dict, Any, Optional, Sequence
+
+import pytz
+from tzlocal import get_localzone
+from mcp.server import Server
+from mcp.server.stdio import stdio_server
+from mcp.types import Tool, TextContent, ImageContent, EmbeddedResource
+
+
+class TimeServer:
+    def __init__(self, local_tz_override: Optional[str] = None):
+        self.local_tz = pytz.timezone(local_tz_override) if local_tz_override else get_localzone()
+
+    def get_current_time(self, timezone_name: str | None = None) -> Dict[str, Any]:
+        """Get current time in specified timezone or local timezone if none specified"""
+        timezone = pytz.timezone(timezone_name) if timezone_name else self.local_tz
+        current_time = datetime.now(timezone)
+        
+        return {
+            "timezone": timezone_name or str(self.local_tz),
+            "time": current_time.strftime("%H:%M %Z"),
+            "date": current_time.strftime("%Y-%m-%d"),
+            "full_datetime": current_time.strftime("%Y-%m-%d %H:%M:%S %Z"),
+            "is_dst": bool(current_time.dst())
+        }
+
+    def convert_time(self, source_tz: str, time_str: str, target_tz: str) -> Dict[str, Any]:
+        """Convert time between timezones"""
+        try:
+            source_timezone = pytz.timezone(source_tz)
+            target_timezone = pytz.timezone(target_tz)
+            
+            # Parse time
+            hour, minute = map(int, time_str.split(":"))
+            if not (0 <= hour <= 23 and 0 <= minute <= 59):
+                raise ValueError
+        except pytz.exceptions.UnknownTimeZoneError as e:
+            raise ValueError(f"Unknown timezone: {str(e)}")
+        except:
+            raise ValueError("Invalid time format. Expected HH:MM (24-hour format)")
+        
+        # Create time in source timezone
+        now = datetime.now(source_timezone)
+        source_time = source_timezone.localize(
+            datetime(now.year, now.month, now.day, hour, minute)
+        )
+        
+        # Convert to target timezone
+        target_time = source_time.astimezone(target_timezone)
+        date_changed = source_time.date() != target_time.date()
+        
+        return {
+            "source": {
+                "timezone": str(source_timezone),
+                "time": source_time.strftime("%H:%M %Z"),
+                "date": source_time.strftime("%Y-%m-%d"),
+                "full_datetime": source_time.strftime("%Y-%m-%d %H:%M:%S %Z")
```

Comment:
Since this is read by an LLM, we should just return isoformat. It`s well understood.